src/platform/ModulesManager.c

```diff
@@ -0,0 +1,696 @@
+                    {
+                        OsConfigLogError(GetPlatformLog(), "Object at index %d is missing '%s'", i, g_objectName);
+                        g_numReportedObjects--;
+                    }
+                    else if (NULL == (reported->object = strdup(reported->object)))
+                    {
+                        OsConfigLogError(GetPlatformLog(), "Failed to allocate memory for object name");
+                        g_numReportedObjects--;
+                    }
+                }
+            }
+        }
+    }
+
+    if (config)
+    {
+        json_value_free(config);
+    }
+}
+
+void AreModulesLoadedAndLoadIfNot(const char* directory, const char* configJson)
+{
+    if (NULL == g_modules)
+    {
+        LoadModules(directory, configJson);
+    }
+}
+
+static void FreeModules(MODULE* modules)
+{
+    MODULE* curr = modules;
+    MODULE* next = NULL;
+
+    while (curr)
+    {
+        next = curr->next;
+        UnloadModule(curr);
+        curr = next;
+    }
+
+    modules = NULL;
+}
+
+static void FreeSessions(SESSION* sessions)
+{
+    SESSION* curr = sessions;
+    SESSION* next = NULL;
+    MODULE_SESSION* moduleSession = NULL;
+
+    while (curr)
+    {
+        next = curr->next;
+
+        while (curr->modules)
+        {
+            moduleSession = curr->modules;
+            curr->modules = moduleSession->next;
+            FREE_MEMORY(moduleSession);
+        }
+
+        FREE_MEMORY(curr->uuid);
+        FREE_MEMORY(curr->client);
+        FREE_MEMORY(curr);
+
+        curr = next;
+    }
+
+    sessions = NULL;
+}
+
+static void FreeReportedObjects(REPORTED_OBJECT* reportedObjects, int numReportedObjects)
+{
+    for (int i = 0; i < numReportedObjects; i++)
+    {
+        FREE_MEMORY(reportedObjects[i].component);
+        FREE_MEMORY(reportedObjects[i].object);
+    }
+
+    FREE_MEMORY(reportedObjects);
+}
+
+void UnloadModules(void)
+{
+    FreeSessions(g_sessions);
+    FreeModules(g_modules);
+    FreeReportedObjects(g_reportedObjects, g_numReportedObjects);
+
+    g_numReportedObjects = 0;
+}
+
+static char* GenerateUuid(void)
+{
+    char* uuid = NULL;
+    static const char uuidTemplate[] = "xxxxxxxx-xxxx-Mxxx-Nxxx-xxxxxxxxxxxx";
+    const char* hex = "0123456789ABCDEF-";
+
+    if (NULL == (uuid = (char*)malloc(UUID_LENGTH + 1)))
+    {
+        return NULL;
+    }
+
+    srand(clock());
+
+    for (int i = 0; i < UUID_LENGTH + 1; i++)
+    {
+        int random = rand() % 16;
+        char c = ' ';
+
+        switch (uuidTemplate[i])
+        {
+        case 'x':
+            c = hex[random];
+            break;
+        case '-':
+            c = '-';
+            break;
+        case 'M':
+            c = hex[(random & 0x03) | 0x08];
+            break;
+        case 'N':
+            c = '4';
+            break;
+        }
+        uuid[i] = (i < UUID_LENGTH) ? c : 0x00;
+    }
```

Comment:
Indent 'case' to right with 4 spaces and remove last 'break' (do you also need a 'default'?)